torii/src/routing.rs

```diff
@@ -151,15 +152,19 @@ pub async fn handle_pending_transactions(
     queue: Arc<Queue>,
     sumeragi: SumeragiHandle,
     pagination: Pagination,
-) -> Result<Scale<Vec<SignedTransaction>>> {
+    payload: TransactionPayload,
+) -> Result<Scale<Option<SignedTransaction>>> {
     let query_response = sumeragi.apply_wsv(|wsv| {
         queue
             .all_transactions(wsv)
             .map(Into::into)
             .paginate(pagination)
-            .collect::<Vec<_>>()
-        // TODO:
-        //.batched(fetch_size)
+            .find(|current_transaction: &SignedTransaction| {
```

Comment:
What if there are multiple matching transactions? Is returning only one of them a sensible thing to do? (I don't know how the MST use this..)

Reply:
The behavior is kinda copied from the client side. It was fine back there so I guessed that that is the way.

Reply:
> What if there are multiple matching transactions? Is returning only the first one of them a sensible thing to do? (I don't know how the MSTs use this..)

no, there shouldn't be. Transactions are deduplicated in the queue

Reply:
I'm sorry, I was wrong here. Transctions can indeed be duplicated in the queue